cmd/intoto-custom-predicate/attest.go

```diff
@@ -158,50 +164,89 @@ run in the context of a Github Actions workflow.`,
 			check(err)
 
 			if attPath != "" {
-				s := sigstore.NewDefaultSigner()
-				att, err := s.Sign(ctx, p)
-				check(err)
+				if local {
+					// use local version
+					s := signing.NewDefaultFulcio()
+					att, err := s.Sign(ctx, p)
+					check(err)
 
-				_, err = s.Upload(ctx, att)
-				check(err)
+					r := sigstore.NewDefaultRekor()
 
-				f, err := getFile(attPath)
-				check(err)
+					_, err = r.Upload(ctx, att)
+					check(err)
 
-				_, err = f.Write(att.Bytes())
-				check(err)
+					f, err := getFile(attPath)
+					check(err)
+
+					_, err = f.Write(att.Bytes())
+					check(err)
+				} else {
+
+					_, err = github.NewOIDCClient()
+					check(err)
+
+					s := sigstore.NewDefaultFulcio()
+					att, err := s.Sign(ctx, p)
+					check(err)
+
+					r := sigstore.NewDefaultRekor()
+
+					_, err = r.Upload(ctx, att)
+					check(err)
+
+					f, err := getFile(attPath)
+					check(err)
+
+					_, err = f.Write(att.Bytes())
+					check(err)
+				}
 
 			}
 		},
 	}
 
-	c.Flags().StringVarP(&attPath, "signature", "g", "attestation.intoto.jsonl", "Path to write the signed attestation.")
-	c.Flags().StringVarP(&subjects, "subjects", "s", "", "Formatted list of subjects in the same format as sha256sum.")
-	c.Flags().StringVarP(&predicateType, "predicateType", "t", "", "Predicate type for intoto statement header.")
+	c.Flags().StringVarP(&attPath, "signature", "g", "attestation.intoto.jsonl", "Path to write the signed attestation")
+	c.Flags().StringVarP(&subjects, "subjects", "s", "", "Formatted list of subjects in the same format as sha256sum")
+	c.Flags().StringVarP(&predicateType, "predicateType", "t", "", "Predicate type for intoto statement header")
 	c.Flags().StringVarP(&predicateFile, "predicateFile", "f", "predicate.json", "Path to retrieve custom predicate to create attestation from")
 	// Pass in SBOM file OR the sha256 and the sbomURI
 	c.Flags().StringVarP(&sbomFile, "sbom", "b", "", "Path to create SBOM predicate")
 	c.Flags().StringVarP(&sbomSha256, "sbomSha256", "d", "", "Sha256 hash the SBOM")
 	c.Flags().StringVarP(&sbomUri, "sbomUri", "u", "", "SBOM Uri if file not provided")
+	c.Flags().StringVarP(&artifactRepo, "art-repo", "a", "No Assertion", "Github repository from which the artifact was built")
+	c.Flags().StringVarP(&artifactRepoCommit, "art-repo-commit", "c", "No Assertion", "Commit of repository from which the artifact was built")
+	c.Flags().StringVarP(&attestationGenRepo, "att-generation-repo", "x", "No Assertion", "Github repository used to generate the attestation")
+	c.Flags().StringVarP(&attestationGenRepoCommit, "att-generation-repo-commit", "y", "No Assertion", "Commit of Github repository used to generate the attestation")
```

Comment:
minor nit: let's make this `NOASSERTION` or `NO_ASSERTION` so its less subject to wrong caps comparison if used programmatically